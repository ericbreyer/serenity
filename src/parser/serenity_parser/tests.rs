#![cfg(test)]
use super::*;

use test_case::test_case;
use insta::{self, assert_debug_snapshot};

#[test_case( "var a: int;", false;"var_decl_simple_type")]
#[test_case( "var a: int", true;"var_decl_simple_type_missing_semicolon")]
#[test_case( "var : int;", true;"var_decl_simple_type_missing_name")]
#[test_case( "var a int;", true;"var_decl_simple_type_missing_colon")]
#[test_case( "var a: int = 5;", false;"var_decl_simple_type_with_init")]
#[test_case( "var a: int = 5", true;"var_decl_simple_type_with_init_missing_semicolon")]
#[test_case( "var a int = 5;", true;"var_decl_simple_type_with_init_missing_colon")]
#[test_case( "var : int = 5;", true;"var_decl_simple_type_with_init_missing_name")]
fn test_parser(snippet: &str, should_fail: bool) {
    let res = SerenityParser::parse(snippet.into(), "test".into());

    let mut settings = insta::Settings::clone_current();
    settings.set_snapshot_suffix(format!("{snippet}"));
    let _guard = settings.bind_to_scope();


    if should_fail {
        assert!(res.is_err());
        let Err(err) = res else { unreachable!() };
        assert_debug_snapshot!(err);
    } else {
        assert!(res.is_ok());
        let Ok(ast) = res else { unreachable!() };
        assert_debug_snapshot!(ast);
    }
}
