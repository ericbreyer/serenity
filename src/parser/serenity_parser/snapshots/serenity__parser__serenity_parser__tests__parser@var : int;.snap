---
source: src/parser/serenity_parser/tests.rs
expression: err
---
[line 1] error at ':' : Expect variable name.
